addons/mail/wizard/mail_compose_message.py

```diff
@@ -595,6 +595,12 @@ def _onchange_template_id(self, template_id, composition_mode, model, res_id):
             default_values = self.with_context(default_composition_mode=composition_mode, default_model=model, default_res_id=res_id).default_get(['composition_mode', 'model', 'res_id', 'parent_id', 'partner_ids', 'subject', 'body', 'email_from', 'reply_to', 'attachment_ids', 'mail_server_id'])
             values = dict((key, default_values[key]) for key in ['subject', 'body', 'partner_ids', 'email_from', 'reply_to', 'attachment_ids', 'mail_server_id'] if key in default_values)
 
+        if template_id:  # Restore default values for fields not updated on template switch (for both "mass_mail" and "comment" modes)
+            fields_to_reset = {'subject', 'body', 'email_from', 'reply_to', 'mail_server_id'} - values.keys()
+            if fields_to_reset:
+                context_dict = {'default_composition_mode': composition_mode, 'default_model': model, 'default_res_id': res_id}
```

Comment:
As we're not getting those (not in `fields_to_reset`), no need to force their default in context

Reply:
so I don't really need to update context right?

```diff
-                values.update(self.with_context(**context_dict).default_get(fields_to_reset))
+                values.update(self.default_get(fields_to_reset))
```